Cython/Compiler/ModuleNode.py

```diff
@@ -3537,9 +3539,14 @@ def generate_type_import_call(self, type, code, import_generator, error_code=Non
                 code.putln("")  # start in new line
             code.putln("#if defined(PYPY_VERSION_NUM) && PYPY_VERSION_NUM < 0x050B0000")
             code.putln('sizeof(%s),' % objstruct)
-            code.putln("#elif CYTHON_COMPILING_IN_LIMITED_API")
-            code.putln('sizeof(%s),' % objstruct)
-            code.putln("#else")
+            if is_builtin_type:
+                # Builtin types are opaque in when the limited API is enabled
+                # and subsequents attempt to access their fields will trigger
+                # compile errors. Skip the struct size check here so things keep
+                # working when a builtin type is imported but not actually used.
```

Comment:
Maybe we should just handle declared builtin types differently, much earlier on. I mean, they don't need to be imported, for example. They are just statically there. This place here seems _very_ late for this distinction.

Reply:
I mean the the problem comes down to the fact that the `ctypedef class xxx [object yyy]:` construct is too powerful for the limited API. It gives cython code the ability to access fields within those types which is impossible to support in a limited API setting.

I think ideally cython should just `#error` when the source file containes such a `ctypedef class` and the user compiled in limited API mode. In practice that wasn't really possible because some modules do `cimport cpython.xxx` files that contain such `ctypedef`s (even though the modules at hand included them for some functions and not for the types).